### ISSUE TITLE
Course - Topic Page

### ISSUE THREAD

**Issue by ecureuill** [solex2006/SELIProject#143]
#114
![image](https://user-images.githubusercontent.com/993369/80646348-ed2f5c00-8a42-11ea-8566-ca89f35fba56.png)

**Comment by ecureuill** [comment]
Meet May 25
@ecureuill @flaviolima @juantenesaca 
**************
1. In couse plan step, teacher selects a model: _Spiral Model_, _Consistent Metaphor_, _Toybox_
   1. Each model has a set of templates
1. in design step, teacher creates the course structure by adding one or more **topics** (course subjects)
   1. For each Topic, list the **resources** he will use
      1. **Materials**, e.g. texts, books, videos, etc
      1. **Tasks**, i.e activities, quiz, forum, problems
           1. For each task, also list **materials**
1. in program step, system should load with 
    1. Organization tab: list of topics and tasks (see wireframes below)
    1. Content Area: will load with a template
       1.  Content Area in **Topics**: using the **Topic Page Template** of the selected model (course plan step)
       1. Content Area in **Activity**: using the **Activity Page Template** of the selected model (course plan step)
       1. Content Area in **Problems**: using the **Problem Page Template** of the selected model (course plan step)
       1. Content Area in **Forum**: using the **Forum Page Template** of the selected model (course plan step)
        
![image](https://user-images.githubusercontent.com/993369/83065635-2f47bf80-a03a-11ea-8b91-a7625b2d3850.png)

![image](https://user-images.githubusercontent.com/993369/83065822-7c2b9600-a03a-11ea-89d7-2b1ca9e78de4.png)

![image](https://user-images.githubusercontent.com/993369/83065887-9e251880-a03a-11ea-87ff-5c830d7753c4.png)

![image](https://user-images.githubusercontent.com/993369/83070833-746fef80-a042-11ea-88a7-ff9c0f3fa154.png)

![image](https://user-images.githubusercontent.com/993369/83105562-4796f900-a091-11ea-9716-790aa98db070.png)

**Comment by ecureuill** [comment]
@juantenesaca @CAndresH  please have a look.

**Comment by juantenesaca** [comment]
@ecureuill Is it necessary to have the option to edit, erase or add units or topics on the navigation tab? What do you think to have this functionalities only in the design step?

**Comment by ecureuill** [comment]
This funcionality is for whom 

> @ecureuill Is it necessary to have the option to edit, erase or add units or topics on the navigation tab? What do you think to have this functionalities only in the design step?

This functionality is exclusive for the "Free" flow.
If teacher selects on Course Plan the Guided option, teacher only manage topics/units on Design step. But if teacher selects on Course Plan the Free option, teacher will manage topics/units on Program step (as it was before instructional design changes)

**Comment by juantenesaca** [comment]
But, although you have selected free, the design step will not disappear as the specifications. So the user will have different information in the design and in the program (number of topics, or if it is on units or topics, the tittle and everything). For this reason I suggest to not have this options on the navigation tab. What do you think?

**Comment by juantenesaca** [comment]
@ecureuill Ok, thanks for the clarification.

**Comment by caroltaus** [comment]
- [ ] Typing errors:

> Should be "Required step"
> ![image](https://user-images.githubusercontent.com/48024795/88579226-19e21800-d020-11ea-9cff-a10c64f13858.png)
> "Requiredaaa"
> ![image](https://user-images.githubusercontent.com/48024795/88579324-3ed68b00-d020-11ea-88fe-b780dc51caed.png)

---------------------------------------------
- [ ] Tasks are not appearing in the organization tab when user creates a course by topics

> ![image](https://user-images.githubusercontent.com/48024795/88579596-aa205d00-d020-11ea-8902-336730cbde37.png)
> ![image](https://user-images.githubusercontent.com/48024795/88579665-cde3a300-d020-11ea-9a65-e807ff3f14a3.png)

----------------------------------------
- [ ] User cannot add/edit/delete in organization tab when in free flow (like described in https://github.com/solex2006/SELIProject/issues/143#issuecomment-647627803 )

**Comment by caroltaus** [comment]
- [ ] There's no field to add instructions like the snackbar asks the user

> ![image](https://user-images.githubusercontent.com/48024795/89221802-465edc80-d5aa-11ea-9192-1d3af0f77366.png)

**Comment by caroltaus** [comment]
- [ ] Snackbar says "audio content" for image content

> ![image](https://user-images.githubusercontent.com/48024795/89829643-51c58100-db31-11ea-8b0c-394ac36d93b4.png)

**Comment by caroltaus** [comment]
- [ ] Responsiveness errors:

modals are not completely shown:
![image](https://user-images.githubusercontent.com/48024795/91088835-15ae1800-e629-11ea-83f3-1000d55f6cca.png)
![image](https://user-images.githubusercontent.com/48024795/91088882-265e8e00-e629-11ea-9220-5331a3449957.png)

buttons on top of each other:
![image](https://user-images.githubusercontent.com/48024795/91088990-49893d80-e629-11ea-97b4-410ca6973949.png)

**Comment by ecureuill** [comment]
Please, move to CODED only when this feature is **fully implemented**: bugs reported in comments and requirements in OP
If you want that we test some partial solution, add a comment here with our @

**Comment by juantenesaca** [comment]
Due the library react-smooth-dnd creates a ghost element when dragging elements. It is not possible to change the cursor icon during this event. I have changed the colors, opacities and background images to solve the drag and drop feedback. Please review the solution.

**Comment by juantenesaca** [comment]
Hello @caroltaus,

About "Keyboard navigation for image and video galleries in previews are not working", this should not be a requirement here, due this has already two issues to develop, number #159 and #160. Plus, It is is not quite important, because program step is in the tutor side.

**Comment by caroltaus** [comment]
@juantenesaca yes, it doesn't need to be accessibe in tutor's side. Please delete, then, the instructions
![image](https://user-images.githubusercontent.com/48024795/96931013-04e81a00-1493-11eb-8991-b0e341bdde64.png)

**Comment by juantenesaca** [comment]
@caroltaus It does not to be deleted, because anyway we have to achieve this feature in image and video gallery issues. My suggestion was that it does not need to be mentioned here in the program, because there are issues dedicated for it.

**Comment by caroltaus** [comment]
@juantenesaca  okay, i understand what you mean now! thank you
